modules/net-vpc-network-firewall/README.md

```diff
@@ -0,0 +1,233 @@
+        direction          = "EGRESS"
+        disabled           = false
+        enable_logging     = true
+        target_secure_tags = ["516738215535"]
+        dest_ip_ranges     = ["192.168.0.0/24"]
+      }
+  }
+}
+
+module "regional_policy" { # regional firewall rules using YAML by defining rules file location
+  source                 = "./fw"
+  regional_policy_name   = "regional-policy-1"
+  project_id             = "my-project"
+  firewall_policy_region = "australia-southeast1"
+  deployment_scope       = "regional"
+  data_folders           = ["./regional-rules"]
+}
+# tftest skip
+```
+
+### Example Configuration Structure using yaml files to create rules
+
+```bash
+├── common
+│   ├── global-rules.yaml
+│   └── regional-rules.yaml
+├── dev
+│   ├── team-a
+│   │   ├── databases.yaml
+│   │   └── regional-rules.yaml
+│   └── team-b
+│       ├── backend.yaml
+│       └── frontend.yaml
+└── prod
+    ├── team-a
+    │   ├── regional-rules.yaml
+    │   └── webb-app-a.yaml
+    └── team-b
+        ├── backend.yaml
+        └── frontend.yaml
+```
+
+### Rule definition format and structure
+
+```yaml
+rule-name: # rule descriptive name
+  disabled: false #`false` or `true`, FW rule is disabled when `true`, default value is `false`
+  description: rule-2 # rules description
+  action: allow # allow or deny
+  direction: INGRESS # INGRESS or EGRESS
+  priority: 1000 # rule priority value, default value is 1000
+  enable_logging: true # Enable rule logging. Default is false
+  src_secure_tags: ["12345678912", ["98765432198"]] # list of source secure tag
+  layer4_configs:
+    - protocol: tcp # protocol, put `all` for any protocol
+      port: ['443', '80', "140-150"] # ports for a specific protocol, keep empty list `[]` for all ports
+  target_service_accounts: # list of target service accounts
+  dest_ip_ranges: # list of destination ranges, should be specified only for `EGRESS` rule
+  src_ip_ranges: # list of source ranges, should be specified only for `INGRESS` rule
+  target_secure_tags: # list of taget secure tag
+  ```
+
+Firewall rules example yaml configuration
+
+```yaml
+rule-1:
+  description: global rule 1
+  action: allow
+  direction: ingress
+  priority: 1000
+  enable_logging: true
+  layer4_configs:
+    - protocol: tcp
+      ports:
+        - 80
+        - 443
+    - protocol: udp
+      ports:
+        - 555
+        - 666
+  src_ip_ranges:
+   - 192.168.1.100/32
+   - 10.10.10.0/24
+  target_secure_tags:
+    - 446253268572
+  src_secure_tags:
+    - 516738215535
+    - 839187618417
+
+rule-2:
+  description: global rule 2
+  action: allow
+  direction: EGRESS
+  priority: 1000
+  enable_logging: true
+  layer4_configs:
+    - protocol: tcp
+      ports:
+        - 80
+        - 443
+    - protocol: udp
+      ports:
+        - 555
+        - 666
+  dest_ip_ranges:
+    - 192.168.0.0/24
+    - 172.16.10.0/24
+  target_secure_tags:
+    - 446253268572
+  
+```
+
+<!-- BEGIN_TF_DOCS -->
```

Comment:
Please make sure to generate this section use the `tfdoc.py` script in the `tools/` folder.